frontend/src/pages/surveys/survey.tsx

```diff
@@ -0,0 +1,52 @@
+import { Header, Navigate } from '#libs/components/components.js';
+import { AppRoute } from '#libs/enums/enums.js';
+import {
+  useAppDispatch,
+  useAppSelector,
+  useCallback,
+} from '#libs/hooks/hooks.js';
+import { type UserAuthResponseDto } from '#packages/users/users.js';
+import { actions as authActions } from '#slices/auth/auth.js';
+
+import { PreferencesStep } from './components/components.js';
+import styles from './styles.module.scss';
+
+const Survey: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const { userId, isSurveyCompleted } = useAppSelector(({ auth }) => ({
+    userId: (auth.authenticatedUser as UserAuthResponseDto).id,
+    isSurveyCompleted: auth.authenticatedUser?.isSurveyCompleted,
```

Comment:
Same as userId

Reply:
I changed destruction not to write type assertion twice

Reply:
@v0ldemar01 I understand, I just did it this way with using only one `as`

![image](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/92298163/3acba925-1f97-4fa9-8912-59fd24e6d33a)